presto-parquet/src/test/java/com/facebook/presto/parquet/writer/TestParquetWriter.java

```diff
@@ -109,27 +114,60 @@ public void testRowGroupFlushInterleavedColumnWriterFallbacks()
         }
     }
 
-    @Test
-    public void testMetadataCreation()
+    @DataProvider(name = "testMetadataCreation")
+    public static Object[][] types() {
+        return new Object[][] {{
+                ImmutableList.of(
+                        TINYINT,
+                        SMALLINT,
+                        BIGINT,
+                        INTEGER,
+                        DOUBLE,
+                        VARCHAR,
+                        BOOLEAN,
+                        SHORT_DECIMAL,
+                        LONG_DECIMAL,
+                        DATE,
+                        TIMESTAMP,
+                        MAP,
+                        ROW),
+                ImmutableList.of(
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.of(LogicalTypeAnnotation.DecimalLogicalTypeAnnotation.class),
+                        Optional.of(LogicalTypeAnnotation.DecimalLogicalTypeAnnotation.class),
+                        Optional.of(LogicalTypeAnnotation.DateLogicalTypeAnnotation.class),
+                        Optional.of(LogicalTypeAnnotation.TimestampLogicalTypeAnnotation.class),
+                        Optional.of(LogicalTypeAnnotation.MapLogicalTypeAnnotation.class),
+                        Optional.empty()),
+                ImmutableList.of(
+                        Optional.of("INT32"),
+                        Optional.of("INT32"),
+                        Optional.of("INT64"),
+                        Optional.of("INT32"),
+                        Optional.of("DOUBLE"),
+                        Optional.of("BINARY"),
+                        Optional.of("BOOLEAN"),
+                        Optional.of("INT64"),
+                        Optional.of("FIXED_LEN_BYTE_ARRAY"),
+                        Optional.of("INT32"),
+                        Optional.of("INT64"),
+                        Optional.empty(),
+                        Optional.empty())
+        }};
+    }
+
+    @Test(dataProvider = "testMetadataCreation")
+    public void testMetadataCreation(List<Type> types, List<Optional<Class<?>>> annotationTypes, List<Optional<String>> primitiveNames)
```

Comment:
Rather than accepting a list of inputs, the signature should accept a single value for each type
```suggestion
    public void testMetadataCreation(Type type, Optional<Class<?>> annotationType, Optional<String> primitiveName)
```